File: source python/Utilities/Savingsystem/Loader.py

class Loader:

    def __init__(self, filepath):
        self.__filepath = filepath
        self.__file = open(filepath, "r")

    def Next(self):
        return self.__file.readline().rstrip()

    def GoTo(self, linePrefix):
        self.__file.close()
        self.__file = open(self.__filepath, "r")
        for line in self.__file:
            line = line.rstrip()
            if line == linePrefix:
                return True
        return False

File: source c++/SavingSystem/Loader.cpp
#include "Loader.h"


Loader::Loader()
{
}

Loader::Loader(const std::string & filepath)
	:	file_(filepath)
{
}

bool Loader::GoTo(const std::string & linePrefix)
{
	file_.seekg(0, std::ios::beg);

	std::string input;
	while (file_ >> input && input != linePrefix)
		file_.ignore(std::numeric_limits<std::streamsize>::max(), '\n');

	return input == linePrefix;
	
}

bool Loader::IsOk()
{
	return file_.is_open();
}

bool Loader::OpenFile(const std::string & filepath)
{
	if (file_.is_open())
		file_.close();
	file_.open(filepath);
	return file_.is_open();
}


int Loader::NextInt()
{
	int result;
	file_ >> result;
	return result;
}

std::string Loader::NextString()
{
	std::string result;
	file_ >> result;
	return result;
}

Loader::~Loader()
{
	if (file_.is_open())
		file_.close();
}
